### PR TITLE
Merge test workflow

### PR DESCRIPTION
**Pull request by abubakkar2024**
- conftest.py added to run tests under certain conditions when to skip and continue running them. All tests fail when unable to collect imports and milvus not enabled at specified port, hence conf code file assists in skipping them when the substring occur in the test code. ( milvus exception)
- Added a github actions workflow to run pytests on merge

### CODE REVIEW

**Review by parkererickson-tg — CHANGES_REQUESTED** [review]
can we gitignore the .idea directory?

Doesn't look like the action worked either

**Comment by abubakkar2024** [comment]
> can we gitignore the .idea directory?
> 
> Doesn't look like the action worked either

will make the changes

File: app/metrics/tg_proxy.py
Comment (parkererickson-tg) [inline app/metrics/tg_proxy.py:1]: Is this rebased with main?

File: app/sync/eventual_consistency_checker.py
Comment (parkererickson-tg) [inline app/sync/eventual_consistency_checker.py:1]: Is this rebased?

**Review by parkererickson-tg — APPROVED** [review]
Confused about ECC and tg_proxy changes - don't want to break these

File: app/sync/eventual_consistency_checker.py
Comment (ahmed-tg) [reply app/sync/eventual_consistency_checker.py:1]: thanks fixed!

**Pull request merged**